src/Compilers/Core/Portable/CodeAnalysisResources.resx

```diff
@@ -522,6 +522,21 @@
     <value>Exception occurred with following context:
 {0}</value>
   </data>
+  <data name="AnonymousTypeMemberAndNamesCountMismatch2" xml:space="preserve">
```

Comment:
I recommend following the pattern the IDE uses.  IN this case, you'd have "_0_and_1_must_have_the_same_length" as the name of the string.  This makes things *super* easy to find when you're trying to go back from a message in an error list to the actual loc string. 

Your call though.